File: py3/944-DeleteColumnsToMakeSorted.py
# 2022-05-12 02:52:28
# Runtime 184ms(15.8%)
# Memory 16.3mb(6.0%)

class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        def isUpward(s):
            for i in range(len(s) - 1):
                if ord(s[i]) > ord(s[i + 1]):
                    return False
            return True
        
        ret = 0
        for i in [[i[j] for i in strs] for j in range(len(strs[0]))]:
            if not isUpward(i):
                ret += 1
        return ret

File: cpp/944-DeleteColumnsToMakeSorted.cpp
// 2022-05-12 13:47:01
// Runtime 40ms(34.0%)
// Memory 11.8mb(92.7%)

class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int ret = 0;
        for (int i = 0; i < strs[0].size(); ++i) {
            for (int j = 0; j < strs.size() - 1; ++j) {
                if (strs[j][i] > strs[j + 1][i]) {
                    ++ret;
                    break;
                }
            }
        }
        return ret;
    }
};
